js/viewmodel.js

```diff
@@ -212,10 +229,11 @@ window.FBE = window.FBE || {};
     function getDefaultDirection(entity) {
         if (entity.type === 'input') { return 2; }
         if (entity.type === 'output') { return 6; }
-        if (/inserter|offshore_pump/.test(entity.name)) { return 6; }
-        if (/splitter|burner_mining/.test(entity.name)) { return 4; }
-        if (/electric_mining|pipe_to_ground|steam_engine|steam_turbine|pump|gate|arithmetic_combinator|decider_combinator/.test(entity.name)) { return 2; }
-        return 0;
+
+        var first = defaultDirectionsByEntityName
+            .filter(function(x){ return x.regex.test(entity.name);})[0];
+
+        return (first && first.direction) || 0;

```

Comment:
still doing this strange conditional that returns a non-boolean

Reply:
Using the logical operators to for short-circuit evaluation is a pretty common idiom, and the simplest way to guard against nulls or provide default values. 

See also: http://eloquentjavascript.net/01_values.html#h_3jN0iK4yKW

In this case, I think a ternary will be clearer, but it'll still be using non-booleans / truthiness